src/ol/renderer/layerrenderer.js

```diff
@@ -95,6 +97,33 @@ ol.renderer.Layer.prototype.handleImageChange = function(event) {
 
 
 /**
+ * Load the image if not already loaded, and manage the
+ * registration/deregistration of the image change listener.
+ * @param {ol.ImageBase} image Image.
+ * @return {boolean} `true` if the image is already loaded, `false`
+ *     otherwise.
+ * @protected
+ */
+ol.renderer.Layer.prototype.loadImage = function(image) {
+  var imageState = image.getState();
+  if (imageState == ol.ImageState.LOADED ||
+      imageState == ol.ImageState.ERROR) {
+    goog.events.unlisten(image, goog.events.EventType.CHANGE,
+        this.handleImageChange_, false, this);
+  } else {
+    if (imageState == ol.ImageState.IDLE) {
+      image.load();
+    }
```

Comment:
But when the same image source is used with multiple maps the maps that won't load the image (because it was loaded by another map) will still register their listener after the image `src` is set.